python-checks/src/main/java/org/sonar/python/checks/cdk/PublicNetworkAccessToCloudResourcesCheck.java

```diff
@@ -19,19 +19,68 @@
  */
 package org.sonar.python.checks.cdk;
 
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
 import org.sonar.check.Rule;
+import org.sonar.plugins.python.api.SubscriptionContext;
+import org.sonar.plugins.python.api.tree.CallExpression;
 
 @Rule(key = "S6329")
 public class PublicNetworkAccessToCloudResourcesCheck extends AbstractCdkResourceCheck {
-  private static final String ERROR_MESSAGE = "Make sure allowing public network access is safe here.";
+
+  private static final String MESSAGE = "Make sure allowing public network access is safe here.";
+  private static final Set<String> SAFE_SUBNET_TYPES = Set.of("ISOLATED", "PRIVATE_ISOLATED", "PRIVATE", "PRIVATE_WITH_NAT");
+  public static final String PUBLICLY_ACCESSIBLE_ARG_NAME = "publicly_accessible";
 
   @Override
   protected void registerFqnConsumer() {
     checkFqn("aws_cdk.aws_dms.CfnReplicationInstance", (subscriptionContext, callExpression) ->
-      CdkUtils.getArgument(subscriptionContext, callExpression, "publicly_accessible").ifPresentOrElse(
-        argument -> argument.addIssueIf(CdkPredicate.isTrue(), ERROR_MESSAGE),
-        () -> subscriptionContext.addIssue(callExpression, ERROR_MESSAGE)
+      CdkUtils.getArgument(subscriptionContext, callExpression, PUBLICLY_ACCESSIBLE_ARG_NAME).ifPresentOrElse(
+        argument -> argument.addIssueIf(CdkPredicate.isTrue(), MESSAGE),
+        () -> subscriptionContext.addIssue(callExpression, MESSAGE)
+      )
+    );
+
+    checkFqn("aws_cdk.aws_rds.DatabaseInstance", PublicNetworkAccessToCloudResourcesCheck::checkDatabaseInstance);
+
+    checkFqn("aws_cdk.aws_rds.CfnDBInstance", (subscriptionContext, callExpression) ->
+      CdkUtils.getArgument(subscriptionContext, callExpression, PUBLICLY_ACCESSIBLE_ARG_NAME).ifPresent(
+        argument -> argument.addIssueIf(CdkPredicate.isTrue(), MESSAGE)
       )
     );
   }
+
+  private static void checkDatabaseInstance(SubscriptionContext ctx, CallExpression call) {
+    Optional<CdkUtils.ExpressionFlow> vpcSubnets = CdkUtils.getArgument(ctx, call, "vpc_subnets");
+
+    Optional<CdkUtils.ExpressionFlow> subnetType = vpcSubnets.flatMap(flow ->
+      CdkUtils.getCall(flow.getLast(), "aws_cdk.aws_ec2.SubnetSelection")
+        .flatMap(subnetSelection -> CdkUtils.getArgument(flow.ctx(), subnetSelection, "subnet_type")));
+
```

Comment:
Not sure if we can call `flow.getLast()` and expect to have the **CallExpression** in every use case, I would recommend to use `flow.getExpression(Predicate<Expression>)` with proper predicate to ensure you are getting the `aws_cdk.aws_ec2.SubnetSelection(...)` **CallExpression**.

Something like :
`CdkUtils.getCall(flow.getExpression(isFqn("aws_cdk.aws_ec2.SubnetSelection").and(expr -> expr.is(Tree.Kind.CALL_EXPR))))`